### ISSUE TITLE
一处模型结构代码和论文相去甚远

### ISSUE THREAD

**Issue by Iloccorb96** [qcf-568/DocTamper#63]
我发现一处模型结构代码和论文相去甚远。如下：
您的models/dtd.py中**第288行**，定义的MID模块中，最后的decoder块，只用了**三个特征图**，与**论文中示意的使用4个特征图不符**
`return self.decoder_convs["x_0_3"](torch.cat((decoder_features["x_0_2"], decoder_features["x_1_2"], decoder_features["x_2_2"]),1))`
倒查代码回去**第274行**，x_0_3层中定义时也有问题，这行显示x_0_3层的输入卷积通道为in_channels[-1]即
`decoder_convs[f"x_{0}_{len(self.in_channels)-1}"] = DecoderBlock(self.in_channels[-1], 0, self.out_channels[-1])`
回到256行`self.in_channels = [encoder_channels[0]] + list(decoder_channels[:-1])`，可知in_channels [-1]为96
根据decoder_block的定义，倒数第三行，会对输入通道截断，即只取了前96个通道。
```
class DecoderBlock(nn.Module):
    def __init__(self,cin,cadd,cout,):
        super().__init__()
        self.cin = (cin + cadd)
        self.cout = cout
        self.conv1 = md.Conv2dReLU(self.cin,self.cout,kernel_size=3,padding=1,use_batchnorm=True)
        self.conv2 = md.Conv2dReLU(self.cout,self.cout,kernel_size=3,padding=1,use_batchnorm=True)

    def forward(self, x1, x2=None):
        x1 = F.interpolate(x1, scale_factor=2.0, mode="nearest")
        if x2 is not None:
            x1 = torch.cat([x1, x2], dim=1)
        x1 = self.conv1(x1[:,:self.cin])
        x1 = self.conv2(x1)
        return x1
```
且您分享的dtd模型文件结构也显示输入为96
 (x_0_3): DecoderBlock(
            (conv1): Conv2dReLU(
              (0): Conv2d(96, 64, kernel_size=(3, 3), stride=(1, 1), padding=(1, 1), bias=False)
根据论文中的模型设计，这一层应该使用**4个特征图的cat，输入通道应该为96*4**
请问开源的模型代码和模型文件，为何是这样的设计，是不小心传错版本了，还是为了简化计算等原因。

**Comment by qcf-568** [comment]
经检查，这里确实存在差异
然而，这个MID解码器模块的核心是纵向和横向的交互，使得多尺度特征充分融合，体现在 [MID](https://github.com/qcf-568/DocTamper/blob/main/models/dtd.py) 的第279~286行。最后输出端的**差异是与设计理念无关的，因此也是对各项结果无影响的** (可通过重新训练来验证，无论是按哪种结果方式一样)。这也类似设计一个新的backbone，方法在backbone本身，而最后输出端是接一层还是两层FC都几乎完全一样。

**Comment by Iloccorb96** [comment]
可以理解，现在所用的特征decoder_features["x_0_2"]，已经是融合了全部尺度信息的特征图，其他三个都是中间过程特征图，确实可以不加上~
